lib/outbox_repository.py

```
@@ -0,0 +1,166 @@
+import json
+
+# from typing import Union, Dict, Any
+
+# Try to import Flask dependencies, fallback to basic logging if not available
+try:
+    from app.logging import get_logger
+    from app.models.database import db
+    from app.models.outbox import Outbox
+    from lib.db import session_guard
+
+    logger = get_logger(__name__)
+    FLASK_AVAILABLE = True
+except ImportError as e:
+    import logging
+
+    logging.basicConfig(level=logging.DEBUG)
+    logger = logging.getLogger(__name__)
+    logger.warning(f"Flask dependencies not available: {e}")
+    FLASK_AVAILABLE = False
+
+
+def _create_update_event_payload(event_dict):
+    host = event_dict.get("host", {})
+    if not host:
+        logger.error("Missing required field 'host' in event data")
+        return None
+
+    metadata = {
+        "localResourceId": host["id"],
+        "apiHref": "https://apiHref.com/",
+        "consoleHref": "https://www.console.com/",
+        "reporterVersion": "1.0",
+    }
+
+    # common = {"workspace_id": event_dict["host"]["groups"][0]["id"]}
+    groups = event_dict.get("host").get("groups")
+    common = {"workspace_id": groups[0]["id"]} if len(groups) > 0 else {}
+
+    reporter = {
+        "satellite_id": host.get("satellite_id", None),
+        "subscription_manager_id": host.get("subscription_manager_id", None),
+        "insights_id": host.get("insights_id", None),
+        "ansible_host": host.get("ansible_host", None),
+    }
+
+    representations = {
+        "metadata": metadata,
+        "common": common,
+        "reporter": reporter,
+    }
+
+    return {
+        "type": "host",
+        "reporterType": "hbi",
+        "reporterInstanceId": "redhat.com",
+        "representations": representations,
+    }
+
+
+def _delete_event_payload(event_dict):
```

Should include a return type hint. Also, similar to the create/update version I think it would make more sense if this function raised an error rather than return None when a required field is missing.